src/awkward/_typetracer.py

```diff
@@ -5,18 +5,23 @@
 import numpy
 
 import awkward.nplike
-import awkward._util
 import awkward as ak
 
 np = ak.nplike.NumpyMetadata.instance()
 
 
+# FIXME: deduplicate this
+# matches include/awkward/common.h
+kMaxInt64 = 9223372036854775806  # 2**63 - 2: see below
+kSliceNone = kMaxInt64 + 1  # for Slice::none()
```

Comment:
Having constants defined once per language is, I think, not bad. The contortions you'd need to pick up common values (e.g. single file source of truth and reading the file in both languages, at compile-time in C++) would be more obscuring than the potential for error in the two copies of the constants.

Maybe they should have symmetric positions in both languages: util.h and _util.py, for instance.

The danger of duplication is that one value can be changed without the maintainer knowing about or remembering the existence of the other value, and then strange bugs occur when some functions pick up one value and other pick up another. But for this sort of constant, that can _never_ happen: we are 100% certain that we'll never change the value of `kMaxInt64`, since it is derived from the bit width of 64-bit numbers.

Reply:
Ah yes, good catch. I forgot that I had to introduce this during refactoring.